# Configuration for the joint state controller
# Each controller allows scripts to access to and modify a joint's state
curiosity_mars_rover:
  joint_state_controller:
    type: "joint_state_controller/JointStateController"
    publish_rate: 50

  # Suspension Position Controllers
  suspension_arm_F_L_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: suspension_arm_F_L_joint
    pid: { p: 2200.0, i: 10.0, d: 10.0 }
  suspension_arm_B_L_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: suspension_arm_B_L_joint
    pid: { p: 4200.0, i: 10.0, d: 10.0 }
  suspension_arm_B2_L_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: suspension_arm_B2_L_joint
    pid: { p: 2200.0, i: 10.0, d: 10.0 }

  suspension_arm_F_R_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: suspension_arm_F_R_joint
    pid: { p: 2200.0, i: 10.0, d: 10.0 }
  suspension_arm_B_R_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: suspension_arm_B_R_joint
    pid: { p: 4200.0, i: 10.0, d: 10.0 }
  suspension_arm_B2_R_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: suspension_arm_B2_R_joint
    pid: { p: 2200.0, i: 10.0, d: 10.0 }

  # Mast Position Controllers
  mast_p_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: mast_p_joint
    pid: { p: 500.0, i: 10.0, d: 1000.0 }
  mast_02_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: mast_02_joint
    pid: { p: 100.0, i: 10.0, d: 100.0 }
  mast_cameras_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: mast_cameras_joint
    pid: { p: 100.0, i: 10.0, d: 100.0 }

  # Arm Position Controllers
  arm_01_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: arm_01_joint
    pid: { p: 500.0, i: 10.0, d: 1000.0 }
  arm_02_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: arm_02_joint
    pid: { p: 500.0, i: 10.0, d: 1000.0 }
  arm_03_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: arm_03_joint
    pid: { p: 500.0, i: 10.0, d: 1000.0 }
  arm_04_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: arm_04_joint
    pid: { p: 100.0, i: 10.0, d: 100.0 }
  arm_tools_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: arm_tools_joint
    pid: { p: 200.0, i: 10.0, d: 100.0 }
